Userland/Libraries/LibC/sys/types.h

```diff
@@ -57,16 +57,21 @@ typedef __WINT_TYPE__ wint_t;
 typedef uint32_t ino_t;
 typedef int64_t off_t;
 
+typedef uint32_t blksize_t;
+typedef uint32_t blkcnt_t;
 typedef uint32_t dev_t;
 typedef uint16_t mode_t;
 typedef uint32_t nlink_t;
-typedef uint32_t blksize_t;
-typedef uint32_t blkcnt_t;
+
 typedef int64_t time_t;
 typedef uint32_t useconds_t;
+typedef uint16_t u_mode_t;
```

Comment:
what's that? It's not present in any `types.h` I've checked and I can find literally 0 results on google..

![image](https://user-images.githubusercontent.com/19366641/115670618-e7184680-a349-11eb-83dd-cedb7d3224c0.png)

Reply:
just saw that `umode_t` is a thing, so probably typo. then why is it next to the time related typedefs though? Also seems to be Linux-specific?

Reply:
> just saw that `umode_t` is a thing, so probably typo. then why is it next to the time related typedefs though? Also seems to be Linux-specific?

I also looked for it under the FreeBSD sources, and found it under the OpenZFS source code. Turns out that part of code uses a Linux compatibility layer, so I'll promptly remove it

Reply:
I got it definitely wrong. I also looked at Haiku because I wanted to check whether I was missing something since Haiku has a BSD compatibility layer and not a Linux one, and since this entire change concerned filesystems in particular, I figured that I must've missed something and added it too with extreme cautiousness as I couldn't find any search results either.

I'll ask their mailing list, just in case, and promptly remove it since every other source seems to say that this is useless.